### PR TITLE
udev: drop workaround for slow read of phys_port_name sysattr

### LINKED ISSUE #21725: Test stability of netdev persistent names
In RHEL we'd like to have some kind of test suite which ensures that we didn't break the netdev naming policy between minor (and major) releases, but I think this would be useful even for upstream.

Since this requires having a machine with all kinds of NICs, I though about using @martinpitt's [umockdev](https://github.com/martinpitt/umockdev/) to *scrape* the necessary sysfs data from various machines and then just mocking them up in a test. Something like this:

```
# Text dump from `udevadm-record`
$ grep -E 'ID_(NET_NAME|OUI)' pyrelight.eth 
E: ID_NET_NAME_MAC=enx8c8caa458e68
E: ID_OUI_FROM_DATABASE=LCFC(HeFei) Electronics Technology co., ltd
E: ID_NET_NAME_PATH=enp0s31f6
E: ID_NET_NAME=enp0s31f6

$ umockdev-run --device pyrelight.eth udevadm test-builtin net_id /sys/class/net/enp0s31f6 2>/dev/null
ID_NET_NAMING_SCHEME=v247
ID_NET_NAME_MAC=enx8c8caa458e68
ID_OUI_FROM_DATABASE=LCFC(HeFei) Electronics Technology co., ltd
ID_NET_NAME_PATH=enp0s31f6
```

This works fine, until you try it on a build with netlink:
```
$ umockdev-run --device pyrelight.eth ~/repos/systemd/build/udevadm test-builtin net_id /sys/class/net/enp0s31f6 2>/dev/null
ID_NET_NAMING_SCHEME=v250
ID_NET_NAME_MAC=en00d86117a86c
ID_OUI_FROM_DATABASE=Micro-Star INTL CO., LTD.
ID_NET_NAME_PATH=enp0s31f6

$ udevadm info /sys/class/net/enp34s0 | grep -E 'ID_(NET_NAME|OUI)' 
E: ID_NET_NAME_MAC=enx00d86117a86c
E: ID_OUI_FROM_DATABASE=Micro-Star INTL CO., LTD.
E: ID_NET_NAME_PATH=enp34s0
E: ID_NET_NAME=enp34s0
```

as with netlink it mixes the sysfs data from the `umockdev` sandbox with the data it gets via netlink from host's udev, which is unfortunate.

1) Is there any way to force `udevadm` (or a similar tool) to use only sysfs?

2) If not (or even if it's possible) is there a better approach that would still allow us to use umockdev?

3) (I had more question on my mind but managed to forget them before I wrote them down)

/cc @yuwata @msekletar (and others, of course)

### PR DESCRIPTION
**Pull request by yuwata**
TL;DR
This effectively reverts 8327fd1b11c5fb6529d46dfb40e2af981ffa8545,
eaba9bb3e69635d2c490c5e1b0d262b763753e1d, and its follow-ups, as the
original issue was already fixed by the kernel side.
    
The original issue that the above commits tried to 'fix' is that reading
phys_port_name triggers a lock in the kernel, hence processing multiple
interfaces at the same time causes extreme slow down.
To workaround the issue, the above commits made several necessary
information retrieved through netlink instead of sysfs attributes.
    
The original issue was already fixed in v5.16 by the series of patches:
https://lore.kernel.org/all/20210928125500.167943-1-atenart@kernel.org/
https://github.com/torvalds/linux/commit/146e5e733310379f51924111068f08a3af0db830
The fix has been already backported to 5.4.160, 5.10.80, 5.14.19, and 5.15.3.
    
When these commits were proposed, it is already claimed that such issue
should be fixed by the kernel side, and udevd should not workaround it.
Neverthless the feature was introduced, as these have theoretical
performance improvement, even if phys_port_name sysattr does not have the
above issue, as in that way udevd can obtain multiple information about
the interface with a single netlink socket operation. See the discussion
in #20744.
    
However, in reality, only `iflink`, `type`, `address`, and `phys_port_name`
attributes from netlink are used in the udev net_id builtin command. Hence,
after the original issue being fixed in the kernel side, there should be
almost no performance improvement for udevd.
Furthermore, combining attributes from netlink and sysfs makes hard to
test net_id builtin. See #21725.
    
Let's drop mostly meaningless code, and make net_id builtin easily testable.
    
Closes #21725.

### CODE REVIEW

**Review by Soumali-123 — APPROVED** [review]

**Comment by yuwata** [comment]
@atenart Let me confirm, the issue for phys_port_name is already fixed by the kernel side, right?

**Comment by yuwata** [comment]
> > @atenart Let me confirm, the issue for phys_port_name is already fixed by the kernel side, right?
> 
> The series you mentioned was only partially applied, as we found the overall approach could not work. In some cases (when the ops for a particular attribute retrieval is not implemented) things are better but the core of the issue is still there. IIRC this alone improved things for the issue that was initially reported, but it is not a solution (the systemd changes where not either).
> 
> IMHO extending the netlink interface would be better than using sysfs, as the latter is not likely to get all the improvements and new features (in addition to other benefits, like having a single call).
> 
> But I get your point, you can't switch 100% to netlink ATM and supporting both interfaces has a cost. The initial argument was to use netlink as it made more sense and optimized things, which as a side effect fixed the "phys_port_name issue". If this has more (maintenance) costs than benefits, I can't really argue with that.

@atenart Thank you for the comments.

If still the performance issue significantly hits some setups, then could you provide a new benchmark? Then, I can implement an alternative workaround for the issue, which will be applied only when a certain environment variable is set, e.g. `UDEV_USE_RTNL_AS_WORKAROUND=1` or so.
